CMakeLists.txt

```diff
@@ -496,6 +502,16 @@ if (BUILD_TESTING)
     file (GLOB TEST_LD_PRELOAD "tests/LD_PRELOAD/*.c")
     add_library(allocator_overrides SHARED ${TEST_LD_PRELOAD})
 
+    set(UNIT_TEST_ENVS S2N_DONT_MLOCK=1)
+    if(TSAN)
+        set(TSAN_SUPPRESSIONS_FILE ${CMAKE_SOURCE_DIR}/tests/.tsan_suppressions)
```

Comment:
Consider a message to tell us if this is on

```suggestion
        message(STATUS "Enabling thread sanitizer.")
        set(TSAN_SUPPRESSIONS_FILE ${CMAKE_SOURCE_DIR}/tests/.tsan_suppressions)

```

Reply:
That would only print the message when running TSAN with BUILD_TESTING also on-- is that what you'd want? In that case, I'd probably make the message more specific like "Running tests with ThreadSanitizer".

I could put the more general "Enabling thread sanitizer" [here](https://github.com/aws/s2n-tls/blob/b79d53d5945632d91a64d4bbf84a1e1b384d96cc/CMakeLists.txt#L225-L226), but I'm not sure that would be necessary since TSAN is a public cmake option.

Reply:
Updated the message; I'm thinking of the case where we're reviewing CI logs and want some confirmation it's on, and BUILD_TESTING is generally on for CodeBuild.

Reply:
I went in to remove the extra newline and realized it would probably be more helpful to just log all the env vars we set, instead of implying we set some with TSAN. Let me know if you disagree.